WooCommerce/Classes/ViewRelated/Products/Edit Product/DefaultProductFormTableViewModel.swift

```diff
@@ -423,6 +425,35 @@ private extension DefaultProductFormTableViewModel {
                                                         title: title,
                                                         details: details)
     }
+
+    // MARK: Linked products only
+
+    func linkedProductsRow(product: ProductFormDataModel, isEditable: Bool) -> ProductFormSection.SettingsRow.ViewModel {
+        let icon = UIImage.linkedProductsImage
+        let title = Localization.linkedProductsTitle
+
+        var linkedProductsDetails = [String]()
+
+        if product.upsellIDs.count <= 1 {
+            linkedProductsDetails.append(String.localizedStringWithFormat(Localization.singularUpsellProductFormat, product.upsellIDs.count))
+        }
+        else {
+            linkedProductsDetails.append(String.localizedStringWithFormat(Localization.pluralUpsellProductsFormat, product.upsellIDs.count))
+        }
+
+        if product.crossSellIDs.count <= 1 {
+            linkedProductsDetails.append(String.localizedStringWithFormat(Localization.singularCrossSellProductFormat, product.crossSellIDs.count))
+        }
+        else {
+            linkedProductsDetails.append(String.localizedStringWithFormat(Localization.pluralCrossSellProductsFormat, product.crossSellIDs.count))
+        }
+        let details = linkedProductsDetails.isEmpty ? nil : linkedProductsDetails.joined(separator: "\n")
```

Comment:
It doesn't look like `linkedProductsDetails.isEmpty` will ever be `true`, right? In any case, I tried this in a playground:

```swift
let names = [String]()
let x = names.joined(separator: "_SEPARATOR_")

print(x)
```

And `x` is just an empty string. So I believe we can simplify this as:

```suggestion
        let details = linkedProductsDetails.joined(separator: "\n")
```

With my other suggestion above, this block could look like:

```swift
let details = [
    Localization.upsellProducts(count: product.upsellIDs.count),
    Localization.crossSellProducts(count: product.crossSellIDs.count),
].joined(separator: "\n")
```